### ISSUE TITLE
Method not found: 'Void Hyperion.SerializerOptions..ctor(Boolean, Boolean, System.Collections.Generic.IEnumerable`1<Hyperion.Surrogate>, System.Collections.Generic.IEnumerable`1<Hyperion.SerializerFactories.ValueSerializerFactory>, System.Collections.Generic.IEnumerable`1<System.Type>)'

### ISSUE THREAD

**Issue by ia64mail** [akkadotnet/Hyperion#71]
I've today updated my application to Akka.Net library version 1.3 from 1.2 and found that it wouldn't work. At startup when Akka system is being created it throws following exception:
```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.MissingMethodException: Method not found: 'Void Hyperion.SerializerOptions..ctor(Boolean, Boolean, System.Collections.Generic.IEnumerable`1<Hyperion.Surrogate>, System.Collections.Generic.IEnumerable`1<Hyperion.SerializerFactories.ValueSerializerFactory>, System.Collections.Generic.IEnumerable`1<System.Type>)'.
   at Akka.Serialization.HyperionSerializer..ctor(ExtendedActorSystem system, HyperionSerializerSettings settings)
   at Akka.Serialization.HyperionSerializer..ctor(ExtendedActorSystem system)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at Akka.Serialization.Serialization..ctor(ExtendedActorSystem system)
   at Akka.Actor.Internal.ActorSystemImpl..ctor(String name, Config config)
   at Akka.Actor.ActorSystem.Create(String name)
   at TestAkka.Program.Main(String[] args) ...
```

I tried the same Akka stack on the new application to eliminate other causes and found that result the same. 

Here is the list of used packages (except `System.*`)
```
  <package id="Akka" version="1.3.0" targetFramework="net46" />
  <package id="Akka.Persistence" version="1.3.0" targetFramework="net46" />
  <package id="Akka.Serialization.Hyperion" version="1.3.0-beta46" targetFramework="net46" />
  <package id="DotNetty.Buffers" version="0.4.6" targetFramework="net46" />
  <package id="DotNetty.Codecs" version="0.4.6" targetFramework="net46" />
  <package id="DotNetty.Common" version="0.4.6" targetFramework="net46" />
  <package id="DotNetty.Handlers" version="0.4.6" targetFramework="net46" />
  <package id="DotNetty.Transport" version="0.4.6" targetFramework="net46" />
  <package id="Google.Protobuf" version="3.3.0" targetFramework="net46" />
  <package id="Hyperion" version="0.9.5" targetFramework="net46" />
  <package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.Extensions.Logging" version="1.1.1" targetFramework="net46" />
  <package id="Microsoft.Extensions.Logging.Abstractions" version="1.1.1" targetFramework="net46" />
  <package id="Microsoft.NETCore.Platforms" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.Win32.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="NETStandard.Library" version="1.6.1" targetFramework="net46" />
  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net46" />
```
And corresponding configuration
```
  <configSections>
    <section name="akka" type="Akka.Configuration.Hocon.AkkaConfigurationSection, Akka" />
  </configSections>

  <akka>
      <hocon>
        <![CDATA[
            akka {
                actor {
                    serializers {
                        wire = "Akka.Serialization.HyperionSerializer, Akka.Serialization.Hyperion"
                    }
                    serialization-bindings {
                        "System.Object" = wire
                    }
                }
            }
        ]]>
      </hocon>
  </akka>
```
The test code is straight forward
```
    class Program
    {
        public static readonly string DEFAULT_SYSTEM_NAME = "test";

        static void Main(string[] args)
        {
                ActorSystem.Create(DEFAULT_SYSTEM_NAME);
        }
    }
```
This code works fine with previous version of the `Hyperion 0.9.2`
```
  <package id="Akka.Serialization.Hyperion" version="1.2.3.43-beta" targetFramework="net46" />
  <package id="Hyperion" version="0.9.2" targetFramework="net46" />
```

**Comment by Danthar** [comment]
Thanks for the detailed report @ia64mail. 
I could not reproduce this issue before. But ill take another stab at it with these settings.

**Comment by Danthar** [comment]
Ok, I still cannot reproduce this. At this point im wondering if its an local environment thing.
Could you check out this project, and see if you have the same problem ?
https://github.com/danthar/hyperiontest

Because that is what i used, and it works for me. (Note that Akka.Serialization.Hyperion latest is 1.3.0-beta47, but that made no difference in my test).

**Comment by Ralf1108** [comment]
Tried yout hyperiontest, still same exception:

System.Reflection.TargetInvocationException occurred
  HResult=0x80131604
  Message=Exception has been thrown by the target of an invocation.
  Source=mscorlib
  StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at Akka.Serialization.Serialization..ctor(ExtendedActorSystem system)
   at Akka.Actor.Internal.ActorSystemImpl..ctor(String name, Config config)
   at Akka.Actor.ActorSystem.Create(String name)
   at HyperionTest.Program.Main(String[] args) in C:\Dev\GitHub\akkaOrig\HyperionTest\HyperionTest\Program.cs:line 27

Inner Exception 1:
MissingMethodException: Method not found: 'Void Hyperion.SerializerOptions..ctor(Boolean, Boolean, System.Collections.Generic.IEnumerable`1<Hyperion.Surrogate>, System.Collections.Generic.IEnumerable`1<Hyperion.SerializerFactories.ValueSerializerFactory>, System.Collections.Generic.IEnumerable`1<System.Type>)'.

**Comment by Danthar** [comment]
@Ralf1108 how do you build&run it ? using dotnet CLI tooling ? or VS ?

**Comment by Ralf1108** [comment]
used

Microsoft Visual Studio Enterprise 2017 
Version 15.2 (26430.16) Release
VisualStudio.15.Release/15.2.0+26430.16
Microsoft .NET Framework
Version 4.6.01590

**Comment by Ralf1108** [comment]
could fix the exception by
- cloning akka project from git
- recompiled Akka.Serialization.Hyperion with latest Hyperion 0.9.5

- uninstalled Akka.Serialization.Hyperion from "hyperiontest"
- referenced recompild Akka.Serialization.Hyperion.dll
- start test program
works!

so just recompile Akka.Serialization.Hyperion and upload the nuget package "Akka.Serialization.Hyperion latest is 1.3.0-beta48" ;-)

**Comment by Ralf1108** [comment]
still getting exception with latest "Akka.Serialization.Hyperion 1.3.0-beta48"

**Comment by Ralf1108** [comment]
Hey guys,

sorry but this issue is not fixed yet, exception still thrown.
The "Akka.Serialization.Hyperion" has to be recompiled and released with the latest Hyperion "0.9.6". Without recompilation the default value for the ctor with the optional parameter is not included in the caller site (here Akka.Serialization.Hyperion).

Releasing a new Hyperion version only fixed the mess with cached older Hyperion versions.
Please check comment from zbynek001: https://github.com/akkadotnet/akka.net/issues/2976#issuecomment-323094050
